GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -553,16 +553,20 @@ private bool NeedsGitStatusMonitor()
 
         private void UICommands_PostRepositoryChanged(object sender, GitUIEventArgs e)
         {
-            this.InvokeAsync(RefreshRevisions).FileAndForget();
+            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await this.SwitchToMainThreadAsync();
+                RefreshRevisions(e.GetRefs);
+            }).FileAndForget();
```

Comment:
`RefreshRevisions` is exectuted by the UI thread now - just delayed. It does several updates which can and should be done in the background.
That's why I suggest to extract an `async` variant of `RefreshRevisions` which switches to the UI thread right before `RevisionGrid.PerformRefreshRevisions(getRefs);`.
A synchronous `RefreshRevisions` which triggers the `async` variant, will be needed, too.

Reply:
(later)

Reply:
The handling is the same, InvokeAsync also switches to main thread. (An argument is needed now so an explicit call is needed).
I believe the event is also executed on the main thread, so no explicit switching.
The only in Refresh not requiring to run on the main thread is the disposing check.
So I am not changing this now.

However, I saw that title in dashboard was not set (actually workin OK anyway, this was a change in another PR that should have set this too).
Also eliminated a special case for refreshing after a plugin has executed (no actual change, no eliminated call).

GetRefs() could be explicitly calculated off the main thread, that would not make any change as the use on the main thread would be waiting anyway.